### ISSUE TITLE
Auto close card view when empty

### ISSUE THREAD

**Issue by poixen** [Cockatrice/Cockatrice#1893]
As a player:
When I scry 1 and move the card to the bottom, I want the window to auto close.
I also want this to happen any other time a similar window would have no cards in it.

**Comment by Nightfirecat** [comment]
:+1: Sounds like a small QOL improvement that would be much appreciated.

That said, I do find it useful sometimes to bring up the windows for GY/exile even when they're empty. If this were to be implemented, I'd like to see the auto-closing only happen when there are 0 cards after removing cards from that view; it shouldn't auto-close if it was opened with 0 cards.

**Comment by Daenyth** [comment]
I agree with Jordan

**Comment by Jinxstands4Jinx** [comment]
I also Agree with Nightfirecat, as I use the side board window in Edh to move away cards that are stolen  from other players who have exited the table for whatever reason. Chucking a card into side boards is better than exile as there are processors now and such so that area can be manipulated.

**Comment by tooomm** [comment]
Very nice suggestion, and perfect sum up by Jordan!

**Comment by Nightfirecat** [comment]
In fact, to expand on what I said earlier, I personally would like it if zones with 0 cards remained open at all times&mdash;that is, even if cards were introduced to the zone, then later removed (ex: cards put into an empty graveyard, then later the graveyard is exiled), the zone would never close on its own. I can understand that this may not be desired by all people, but that's how it would ideally behave in my opinion.

**Comment by skwerlman** [comment]
@Nightfirecat make it an option, then?

**Comment by tooomm** [comment]
Trying to keep it simple... what about this?

if zone is empty (on open) --> never close zone
if zone is not empty (on open) --> close zone after all cards happen to get removed

**Comment by Nightfirecat** [comment]
@tooomm Either option would be good as far as my preferences are concerned, but I'd like to hear from others as well.

**Comment by Daenyth** [comment]
We can just have it close when it goes from having cards to not having cards

**Comment by tooomm** [comment]
@Daenyth this isn't sufficient regarding Jordan's needs.

Imagine this:
He'll open the graveyard of his opponent on game start to better track it (on a second screen for example). Some cards enter and then there is a huge delve and the graveyard turns empty again. He still wants to have it open for further tracking of the next cards without opening it again (even if empty at the moment).

That's why I mentioned to decide the "type" **on open** (close yes/no).
But there are some edge cases, for example if you open the view too late (there are already cards in that zone) or you happen to close it in between by fault... you'll always end up in the "close on empty: yes" type view and can't change it!
<br>
Therefore I pretty much like the tickbox idea, because most people just adjust this once to fit their needs (should be saved for sure) and we have tickboxes in other zone views already. (e.g. library) 
So it's nothing completely new to people and they are used to them.

**Comment by Daenyth** [comment]
Stepping back though, is it really that hard to just close it when you want it closed? Adding a checkbox adds yet another decision players need to make every time, it inches the app toward being more complex and it's an ongoing cognitive overhead for everyone doing it all the time.

I think the current behavior is fine.

**Comment by Daenyth** [comment]
Or maybe we can come up with a consistent rule like, if the view is for a zone you don't own, never auto close. If the view is for a zone you own and the view goes from having 1 card to 0, auto close. Or maybe if an action you take on a card in a view for a zone you own causes the view to go to zero cards, close the view

**Comment by Daenyth** [comment]
I want to avoid as much as possible adding configurations and switches because it makes the app harder to use

**Comment by poixen** [comment]
If you would move a card from a view, leaving 0 cards in that view, auto close. For all other situations, leave open.

**Comment by poixen** [comment]
@Deanyth it's not hard to close when you want it closed. But if you are a moderate to heavy player you can be performing get this one action 100s of times a day. And with the new mulligan scry rule it makes it even more so.

**Comment by tooomm** [comment]
>If **you would move** a card from a view, leaving 0 cards in that view, auto close. For all other situations, leave open. 

That sounds very smart! :+1:

**Comment by Daenyth** [comment]
That's roughly what I was saying. (so I agree with it ;) )
